src/modules/mim/firewall.json

```diff
@@ -1,41 +1,305 @@
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "accept",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "reject",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "drop",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "type",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "inbound",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "outbound",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "forward",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "protocol",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "all",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "tcp",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "udp",
+                      "enumValue": 3
+                    },
+                    {
+                      "name": "icmp",
+                      "enumValue": 4
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "source",
+                "schema": "string"
+              },
+              {
+                "name": "sourcePort",
+                "schema": "integer"
+              },
+              {
+                "name": "destination",
+                "schema": "string"
+              },
+              {
+                "name": "destinationPort",
+                "schema": "integer"
+              },
+              {
+                "name": "interface",
+                "schema": "string"
+              },
+              {
+                "name": "destinationInterface",
+                "schema": "string"
+              },
+              {
+                "name": "state",
+                "schema": {
+                  "type": "array",
+                  "elementSchema": "string"
+                }
+              }
```

Comment:
Do we really need this?

Reply:
This is used to pass symbols to the iptables state module (`established`, `related` etc...). We can omit this, but it will limit the types of rules that can be set.

Reply:
Let's omit the state array

Reply:
Removed for simplification